pkg/kdump/kdump-view.jsx

```diff
@@ -41,15 +41,78 @@ import { Tooltip, TooltipPosition } from "@patternfly/react-core/dist/esm/compon
 import { OutlinedQuestionCircleIcon } from "@patternfly/react-icons";
 
 import { useDialogs, DialogsContext } from "dialogs.jsx";
+import { read_os_release } from "os-release.js";
 import { fmt_to_fragments } from 'utils.jsx';
 import { show_modal_dialog } from "cockpit-components-dialog.jsx";
 import { FormHelper } from "cockpit-components-form-helper";
 import { ModalError } from 'cockpit-components-inline-notification.jsx';
 import { PrivilegedButton } from "cockpit-components-privileged.jsx";
+import { ModificationsExportDialog } from "cockpit-components-modifications.jsx";
 
 const _ = cockpit.gettext;
 const DEFAULT_KDUMP_PATH = "/var/crash";
 
+const exportAnsibleTask = (settings, os_release) => {
+    const target = Object.keys(settings.targets)[0];
+    const targetSettings = settings.targets[target];
+    const kdump_core_collector = settings.core_collector;
+
+    let role_name = "linux-system-roles";
+    if (os_release.NAME === "RHEL" || os_release.ID_LIKE?.includes('rhel')) {
+        role_name = "rhel-system-roles";
+    }
+
+    let ansible = `
+---
+# Also available via https://galaxy.ansible.com/ui/standalone/roles/linux-system-roles/kdump/
+- name: install ${role_name}
+  package:
+    name: ${role_name}
+    state: present
+  delegate_to: 127.0.0.1
+  become: true
+- name: run kdump system role
+  include_role:
+    name: ${role_name}.kdump
+  vars:
+    kdump_path: ${targetSettings.path || DEFAULT_KDUMP_PATH}
+    kdump_core_collector: ${kdump_core_collector}`;
+
+    if (target === "ssh") {
+        // HACK: we should not have to specify kdump_ssh_user and kdump_ssh_user as it is in kdump_target.location
+        // https://github.com/linux-system-roles/kdump/issues/184
+        let ssh_user;
+        let ssh_server;
+        const parts = targetSettings.server.split('@');
+        if (parts.length === 1) {
+            ssh_user = "root";
+            ssh_server = parts[0];
+        } else if (parts.length === 2) {
+            ssh_user = parts[0];
+            ssh_server = parts[1];
+        } else {
+            throw new Error("ssh server contains two @ symbols");
+        }
+        ansible += `
+    kdump_target:
+      type: ssh
+    kdump_sshkey: ${targetSettings.sshkey}
+    kdump_ssh_server: ${ssh_server}
+    kdump_ssh_user: ${ssh_user}`;
+    } else if (target === "nfs") {
+        ansible += `
+    kdump_target:
+      type: nfs
+      location: ${targetSettings.server}:${targetSettings.export}
+`;
+    } else if (target !== "local") {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19430-20231124-103105-864d376a-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)